scenes/prefabs/Dialogue/Dialogue.gd

```diff
@@ -0,0 +1,99 @@
+extends CanvasLayer
+
+export var dialogue_path = "res://res/dialogue/End - auto.csv"
+onready var dialogue_resource = load(dialogue_path)
+
+var _current_event = -1
+var _left_character_sprite = null
+var _right_character_sprite = null
+
+var _is_finished = false
+
+onready var timer = $Timer
+onready var label = $ColorRect/Label
+onready var texture_character_left = $CharacterLeftCanvas/CharacterLeft
+onready var canvas_character_left = $CharacterLeftCanvas
+onready var texture_character_right = $CharacterRightCanvas/CharacterRight
+onready var canvas_character_right = $CharacterRightCanvas
+onready var dialogue_box = $ColorRect
+
+var _active_sprite_y = 0
+var _active_modulate_a = 1
+var _inactive_sprite_y = 100
+var _inactive_modulate_a = 0.5
+
+var _finish_y = 750
+
+var interpolate_speed = 15
+var _left_character_desired_y = _active_sprite_y
+var _left_character_modulate_a = 1
+var _right_character_desired_y = _active_sprite_y
+var _right_character_modulate_a = 1
+
+var _mouse_hover = false
+
+func _ready():
+	_proceed_dialogue()
+	GameManager.SetState(GameManager.GameState.DIALOGUE)
+
+func _process(delta):
+	if Input.is_action_just_pressed("progress_dialogue") or (_mouse_hover and Input.is_action_just_pressed("interact")):
+		_proceed_dialogue()
+	_animate_sprites(delta)
+
+func _animate_sprites(delta):
+	if _is_finished:
+		dialogue_box.modulate.a += (0 - dialogue_box.modulate.a) * interpolate_speed * delta
+	canvas_character_left.offset.y += (_left_character_desired_y - canvas_character_left.offset.y) * interpolate_speed * delta
+	texture_character_left.modulate.a += (_left_character_modulate_a - texture_character_left.modulate.a) * interpolate_speed * delta
+	canvas_character_right.offset.y += (_right_character_desired_y - canvas_character_right.offset.y) * interpolate_speed * delta
+	texture_character_right.modulate.a += (_right_character_modulate_a - texture_character_right.modulate.a) * interpolate_speed * delta
+
+func _proceed_dialogue():
+	_current_event += 1
+	var number_of_dialogue_events = dialogue_resource.events.size()
+	if _current_event >= number_of_dialogue_events:
+		GameManager.SetState(GameManager.GameState.PLAY)
+		timer.start()
+		label.visible = false
+		_is_finished = true
+		interpolate_speed = 5
+		_left_character_desired_y = _finish_y
+		_left_character_modulate_a = 0
+		_right_character_desired_y = _finish_y
+		_right_character_modulate_a = 0
```

Comment:
Wouldn't be useful to put in a function? If you want to get off dialogue other than getting to the end of it.